apis/schema/v1alpha1/mariadbdatabase_webhook.go

```diff
@@ -0,0 +1,223 @@
+		allErrs = append(allErrs, err)
+	}
+	if err := in.validateMariaDBDatabaseConfig(); err != nil {
+		allErrs = append(allErrs, err)
+	}
+	if err := in.validateMariaDBDatabaseNamespace(); err != nil {
+		allErrs = append(allErrs, err)
+	}
+	if err := in.validateMariaDBDatabaseName(); err != nil {
+		allErrs = append(allErrs, err)
+	}
+	if len(allErrs) == 0 {
+		return nil
+	}
+	return apierrors.NewInvalid(schema.GroupKind{Group: "schema.kubedb.com", Kind: "MariaDBDatabase"}, in.Name, allErrs)
+}
+
+func (in *MariaDBDatabase) validateInitailizationSchema() *field.Error {
+	path := field.NewPath("spec.init")
+	if in.Spec.Init != nil {
+		if in.Spec.Init.Script != nil && in.Spec.Init.Snapshot != nil {
+			return field.Invalid(path, in.Name, `cannot initialize database using both restore and initSpec`)
+		}
+	}
+	return nil
+}
+
+func (in *MariaDBDatabase) validateMariaDBDatabaseConfig() *field.Error {
+	path := field.NewPath("spec").Child("database.config").Child("name")
+	name := in.Spec.Database.Config.Name
+	if name == SYSDatabase {
+		return field.Invalid(path, in.Name, `cannot use "sys" as the database name`)
+	}
+	if name == "performance_schema" {
+		return field.Invalid(path, in.Name, `cannot use "performance_schema" as the database name`)
+	}
+	if name == "mysql" {
+		return field.Invalid(path, in.Name, `cannot use "mysql" as the database name`)
+	}
+	if name == DatabaseForEntry {
+		return field.Invalid(path, in.Name, `cannot use "kubedb_system" as the database name`)
+	}
+	if name == "information_schema" {
+		return field.Invalid(path, in.Name, `cannot use "information_schema" as the database name`)
+	}
+	if name == DatabaseNameAdmin {
+		return field.Invalid(path, in.Name, `cannot use "admin" as the database name`)
+	}
+	if name == DatabaseNameConfig {
+		return field.Invalid(path, in.Name, `cannot use "config" as the database name`)
+	}
+	return nil
+}
+
+func (in *MariaDBDatabase) validateMariaDBDatabaseNamespace() *field.Error {
+	path := field.NewPath("metadata").Child("namespace")
+	ns := in.ObjectMeta.Namespace
+	if ns == "cert-manager" {
+		return field.Invalid(path, in.Name, `cannot use namespace "cert-manager" to create schema manager`)
+	}
+	if ns == "kube-system" {
+		return field.Invalid(path, in.Name, `cannot use namespace "kube-system" to create schema manager`)
+	}
+	if ns == "kubedb-system" {
+		return field.Invalid(path, in.Name, `cannot use namespace "kubedb-system" to create schema manager`)
+	}
+	if ns == "kubedb" {
+		return field.Invalid(path, in.Name, `cannot use namespace "kubedb" to create schema manager`)
+	}
+	if ns == "kubevault" {
+		return field.Invalid(path, in.Name, `cannot use namespace "kubevault" to create schema manager`)
+	}
+	if ns == "local-path-storage" {
+		return field.Invalid(path, in.Name, `cannot use namespace "local-path-storage" to create schema manager`)
+	}
+	return nil
+}
+
+func (in *MariaDBDatabase) validateMariaDBDatabaseName() *field.Error {
```

Comment:
Remove Lenght validation, use the default validation

Reply:
removed